### DISCUSSION TITLE
Timed out waiting for copy response

### DISCUSSION THREAD

**Discussion by imtrobin** [rustdesk/rustdesk#8682] [General]
### Bug Description

When the remote desktop source does a copy (windows Ctrl-C), the local viewer (me) shows an error "Time out waiting for copy response". It does not happen always.

### How to Reproduce

Remote desktop source does a Copy (Ctrl-C)

### Expected Behavior

Nothing happens

### Operating system(s) on local side and remote side

windows 10 both

### RustDesk Version(s) on local side and remote side

1.1.25

### Screenshots

https://ibb.co/4svNmpS

### Additional Context

It happens sometimes, even on older rustdesk versions. Note, it is not local machine doing the copy, it is the remote machine I'm viewing.

**Comment by rustdesk** [comment]
@fufesou

**Comment by rustdesk** [comment]
Only happend when there is file pasting dialog in the screenshot?

**Reply by imtrobin** [reply]
it happens sometimes only. Sorry, we are using 1.2.6, but this issue happens even on 1.24 nightlies many months ago

**Comment by fufesou** [comment]
@imtrobin Hi, thanks for your feedback.

"Time out waiting for copy response" means a data chunk cannot be received after 30 seconds. Maybe the message is lost or there's a deadlock?

1. Do you use self-host server?
2. Can you "copy&paster" again without reconnect if you meet this error? Or you have to disconnect and connect, then "copy&paster"?

Can you please also backup the logs and report the issue here when you see this error the next time?

There're some logs about the file copy&paste steps.

The logs location
```
Client side or portable: %appdata%\rustdesk\log
Server side (installed):  C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\log
```

The logs looks like 
```
client_file_contents_request called, data ...
client_file_contents_response called ...
```

**Reply by imtrobin** [reply]
1. no, using rustdesk server.
2. The copy and paste is not done by me. It is the user on the machine I'm controlling.

I'm using portable for both server/client. I will get the log if it happens again.

**Reply by fufesou** [reply]
Thanks for you kind feedback.

> The copy and paste is not done by me. It is the user on the machine I'm controlling.

Please also confirm if it needs reconnection to make "copy&paste" work. So I can know if there's a dead lock.

**Reply by imtrobin** [reply]
I couldn't find the log file. There is a AppData\Roaming\RustDesk\log but it seems to be a old install, I'm using portable now

**Reply by fufesou** [reply]
Yes, logs are located here ` %appdata%\rustdesk\log` for portable version.

You can confirm if it needs reconnection to make "copy&paste" work the next time when you face this issue.

**Comment by imtrobin** [comment]
I deleted all the files in the log folder and I connected rustdesk again. there is no log file created in  %appdata%\rustdesk\log

**Reply by fufesou** [reply]
It's wired.

Do you have everything https://www.voidtools.com/  installed? Can you please search where `rustdesk_rCURRENT.log` is?

**Reply by imtrobin** [reply]
Hmm, there is no rustdesk_rCURRENT.log

**Reply by fufesou** [reply]
`rustdesk_rCURRENT.log` seems only created on process starting.

If you have a runing RustDesk, and the delete the log files, it may not be created again.

Can you please restart RustDesk to confirm if the log files are created?

**Reply by fufesou** [reply]
I've downloaded the file. You can remove it now.

**Reply by fufesou** [reply]
Hi, I've checked the log file.

There are:
1. 8 `client_format_data_request called` and 8 `server_format_data_response called, conn_id 1, msg_flags: 1, return 0`.
2. 2 `client_file_contents_request called` and 2 `server_file_contents_response called, conn_id ..., return 0`

It seems there should be no "Timed out waiting for copy response" in this connection?

**Reply by imtrobin** [reply]
Perhaps not, maybe this is with rustdesk installed on server. Now, I don't seem to get the Timed out anymore.
